product_supplierinfo_for_customer_sale_quick/models/product_product.py

```diff
@@ -0,0 +1,38 @@
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    product_customer_code = fields.Char(
+        string="Product Customer Code",
+        compute="_compute_product_customer_info",
+        search="_search_customer_supplierinfo_data",
+        readonly=True,
+    )
+
+    product_customer_name = fields.Char(
+        string="Product Customer Name",
+        compute="_compute_product_customer_info",
+        search="_search_customer_supplierinfo_data",
+        readonly=True,
+    )
+
+    @api.model
+    def _search_customer_supplierinfo_data(self, operator, value):
+        products = self.env["product.product"].search(
+            [
+                "|",
+                ("product_customer_code", operator, value),
+                ("product_customer_name", operator, value),
+            ]
+        )
```

Comment:
**issue**: The two fields `product_customer_code` and `product_customer_name` are not stored (and they shouldn't be); that means they're calculated  in memory every time they need to be accessed.

Unfortunately, `search` defers the operation to the database, that cannot execute it because those fields do not exist as columns in the DB.

There's also a recursion error here because this method is called to perform the search. When the `search()` method is called here, Odoo encounters the same field again, and calls this method again since Odoo has been told it to use this method to perform a search on these fields.

My suggestion: 
- only associate this method with `product_customer_name`
- rewrite the search so it doesn't use either of these two fields. You may have to search directly on product.customerinfo` something like: 
```
self.env["product.customerinfo"].search([
            ('partner_id', '=', ...),
            "|",
                ('product_code', 'operator', ...),
                ('product_name', 'operator', ...),
        ])
```
- When adding the filter in xml, only add the filter for `product_customer_name`

Functional question: you add the filter to `product.product`, but these fields do not always have a value. They're only given a value if there's a `partner_id` in the context. What should happen if the user searches on these fields in a "normal" view, without `partner_id` in the context?

Reply:
I have implemented the changes. For the functional question, I think it should retrieve nothing, so i will have no output for that search. Or maybe it will thrown an error.